plugin/src/main/java/org/gradle/testretry/internal/executer/framework/TestFrameworkStrategy.java

```diff
@@ -25,25 +26,52 @@
 import org.gradle.util.GradleVersion;
 
 import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Iterator;
+import java.util.regex.Pattern;
 
 /**
  * Instances are scoped to a test task execution and are reused between rounds.
  */
 public interface TestFrameworkStrategy {
 
+    Pattern SPOCK2_CORE_JAR_NAME_PATTERN = Pattern.compile(".*/spock-core-2.*\\.jar");
+
     @Nullable
-    static TestFrameworkStrategy of(TestFramework testFramework) {
+    static TestFrameworkStrategy of(JvmTestExecutionSpec spec) {
+        TestFramework testFramework = spec.getTestFramework();
         if (testFramework instanceof JUnitTestFramework) {
             return new JunitTestFrameworkStrategy();
         } else if (testFramework instanceof JUnitPlatformTestFramework) {
-            return new Junit5TestFrameworkStrategy();
+            return new Junit5TestFrameworkStrategy(isSpock2Used(spec));
         } else if (testFramework instanceof TestNGTestFramework) {
             return new TestNgTestFrameworkStrategy();
         } else {
             return null;
         }
     }
 
+    static boolean isSpock2Used(JvmTestExecutionSpec spec) {
+        return isSpock2JarOnPath(spec.getClasspath()) || isSpock2JarOnPath(spec.getModulePath());
+    }
+
+    static boolean isSpock2JarOnPath(Iterable<? extends File> path) {
+        Iterator<? extends File> classpathIterator = path.iterator();
+        boolean spock2CoreDependencyFound = false;
+        while (classpathIterator.hasNext() && !spock2CoreDependencyFound) {
+            File file = classpathIterator.next();
+            if (isSpock2CoreJar(file)) {
+                spock2CoreDependencyFound = true;
+            }
+        }
+
+        return spock2CoreDependencyFound;
```

Comment:
```suggestion
        StreamSupport.stream(path.spliterator(), false)
            .anyMatch(this::isSpock2CoreJar);
```

Reply:
TIL. Done in https://github.com/gradle/test-retry-gradle-plugin/pull/202/commits/e03e80d0e4981334234d61187dc28f19da1ededb